File: runtime/org.eclipse.etrice.runtime.cpp/src/common/modelbase/EventWithDataMessage.cpp
/*
 * EventWithDataMessage.cpp
 *
 *  Created on: 29.08.2012
 *      Author: karlitsc
 */

#include "EventWithDataMessage.h"
#include <sstream>

namespace etRuntime {

EventWithDataMessage::~EventWithDataMessage() {
}

std::string EventWithDataMessage::toString() {
	std::stringstream strm;
	strm << "EventWithDataMessage(" << getAddress().toString() << ", evt=" << getEvtId() + ", m_data=" << dataToString() << ")";
	return strm.str();
}

std::string EventWithDataMessage::dataToString() {
	if (m_data == 0) {
		return "()";
	}
	//TODO: how to do this with all kinds of data??
	//return "(" + m_data + ")";
	return "()";
}

} /* namespace etRuntime */

File: runtime/org.eclipse.etrice.runtime.java/src/org/eclipse/etrice/runtime/java/modelbase/EventWithDataMessage.java
/*******************************************************************************
 * Copyright (c) 2010 protos software gmbh (http://www.protos.de).
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *******************************************************************************/

package org.eclipse.etrice.runtime.java.modelbase;

import org.eclipse.etrice.runtime.java.messaging.Address;

/**
 * An event message carrying an arbitrary number of objects.
 * 
 * @author Henrik Rentz-Reichert
 *
 */
public class EventWithDataMessage extends EventMessage {

	private Object data = null;

	public EventWithDataMessage(Address address, int evtId, Object data) {
		super(address, evtId);
		this.data = data;
	}

	public Object getData() {
		return data;
	}
	
	@Override
	public String toString() {
		return "EventWithDataMessage("+getAddress()+", evt="+getEvtId()+", data="+dataToString()+")";
	}
	
	public String dataToString() {
		if (data==null)
			return "()";
		
		return "("+data+")";
	}
}
